extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -0,0 +1,325 @@
+    private static final Set<String> USUAL_FEATURES = Sets.newHashSet(FEATURE_CONDITIONS, FEATURE_FORECAST10DAY);
+
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+
+    private WeatherUndergroundJsonData weatherData;
+
+    private ScheduledFuture<?> refreshJob;
+
+    private Gson gson;
+
+    public WeatherUndergroundHandler(Thing thing) {
+        super(thing);
+
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing WeatherUnderground handler.");
+
+        WeatherUndergroundConfiguration config = getConfigAs(WeatherUndergroundConfiguration.class);
+        logger.debug("config apikey = {}", config.apikey);
+        logger.debug("config location = {}", config.location);
+        logger.debug("config language = {}", config.language);
+        logger.debug("config refresh = {}", config.refresh);
+
+        boolean validConfig = true;
+        String errors = "";
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errors += " Parameter 'apikey' must be configured.";
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.location) == null) {
+            errors += " Parameter 'location' must be configured.";
+            validConfig = false;
+        }
+        if (config.language != null) {
+            String lang = StringUtils.trimToEmpty(config.language);
+            if (lang.length() != 2) {
+                errors += " Parameter 'language' must be 2 letters.";
+                validConfig = false;
+            }
+        }
+        if (config.refresh != null && config.refresh < 5) {
+            errors += " Parameter 'refresh' must be at least 5 minutes.";
+            validConfig = false;
+        }
+        errors = errors.trim();
+
+        if (validConfig) {
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+        } else {
+            logger.info("Disabling thing '{}': {}", getThing().getUID(), errors);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errors);
+        }
+    }
+
+    /**
+     * Start the job refreshing the weather data
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request new weather data to the Weather Underground service
+                        updateWeatherData();
+
+                        // Update all channels from the updated weather data
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId());
+                        }
+                    } catch (Exception e) {
+                        logger.debug("Exception occurred during execution: {}", e.getMessage());
```

Comment:
Such catch exception is required to avoid stopping the task scheduling. We have a general issue opened for a better handling of exceptions in scheduled tasks.

The change of status is already done in updateWeatherData(). There is already some catch exceptions in updateWeatherData(), so when an exception is catched here, it should be a kind of bug. I have added the stack trace.